pkg/client/userd/trafficmgr/traffic_manager.go

```diff
@@ -572,12 +585,25 @@ nextIs:
 			}
 		}
 	}
+
 	aMap := make(map[string]*manager.AgentInfo)
-	for _, ns := range nss {
-		for k, v := range tm.getCurrentAgentsInNamespace(ns) {
-			aMap[k] = v
+	if allNamespaces && filter >= connector.ListRequest_INSTALLED_AGENTS {
+		// add all agents to agent map and namespaces
+		for _, v := range tm.getCurrentAgents() {
+			aMap[v.Name] = v
+			if _, ok := nsmap[v.Namespace]; !ok {
+				nsmap[v.Namespace] = true
+				nss = append(nss, v.Namespace)
```

Comment:
I really don't like this part. Wasn't the intention to capture all current intercepts? If so, then why are we scanning *all* namespaces where there's an agent? Please note that this can be agents that in no way are related to the current user.